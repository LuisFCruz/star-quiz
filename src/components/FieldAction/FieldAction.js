import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { TextField, withStyles, Button } from '@material-ui/core';

const styles = {
  container: {
    display: 'flex',
    alignItems: 'inherit',
  },
  button: {
    marginLeft: '5px',
    minWidth: 'auto',
  }
};

export class FieldAction extends Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    onConfirm: PropTypes.func.isRequired,
    placeholder: PropTypes.string,
  }

  constructor() {
    super();

    this.state = {
      value: ''
    };
  }

  handleConfirmClick = () => {
    const { value } = this.state;
    this.props.onConfirm(value);
  }

  handleChange = ({target: {value = ''}}) => {
    this.setState({ value });
  }

  render() {
    const { classes, placeholder = '' } = this.props;
    return (
      <div className={classes.container}>
        <TextField
          placeholder={placeholder}
          onChange={this.handleChange}
        />
        <Button
          color="primary"
          variant="contained"
          className={classes.button}
          onClick={this.handleConfirmClick}
        >ok</Button>
      </div>
    )
  }
}

export default withStyles(styles)(FieldAction);
