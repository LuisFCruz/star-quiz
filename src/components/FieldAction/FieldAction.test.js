import React from 'react';
import { shallow } from 'enzyme';

import { FieldAction } from './FieldAction';

describe('<FieldAction />', () => {
  let component;
  let wrapper;
  const props = {
    classes: {
      button: 'button',
    },
    onConfirm: jest.fn(),
  };

  beforeEach(() => {
    wrapper = shallow(<FieldAction {...props} />);
    component  = wrapper.instance();
  });

  
  test('smoke test', () => {
    expect(FieldAction).toBeDefined();
    expect(component).toBeInstanceOf(FieldAction);
  });
  
  test('render', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should have call handleChange()', () => {
    const handleSpy = jest.fn();
    component.handleChange = handleSpy;
    component.forceUpdate();

    const input = wrapper.find('TextField');
    expect(input).toHaveLength(1);
    input.simulate('change', { target: { value: 'value changed'  } });
    expect(handleSpy).toHaveBeenCalled();
  });
  
  test('should have call handleConfirmClick()', () => {    
    const handleSpy = jest.fn();
    component.handleConfirmClick = handleSpy;
    component.forceUpdate();

    const button = wrapper.find('.button');
    
    expect(button).toHaveLength(1);
    button.simulate('click');
    expect(handleSpy).toHaveBeenCalled();
  });

  test('should have changed value', () => {
    component.handleChange({ target:  { value: 'value changed'  } });
    expect(wrapper.state('value')).toBe('value changed');
  });
});
