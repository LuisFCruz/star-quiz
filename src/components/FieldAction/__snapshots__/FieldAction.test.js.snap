// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FieldAction /> render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FieldAction
    classes={
      Object {
        "button": "button",
        "onConfirm": [MockFunction],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <TextField
          onChange={[Function]}
          placeholder=""
          required={false}
          select={false}
          variant="standard"
        />,
        <WithStyles(Button)
          className="button"
          color="primary"
          onClick={[Function]}
          variant="contained"
        >
          ok
        </WithStyles(Button)>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onChange": [Function],
          "placeholder": "",
          "required": false,
          "select": false,
          "variant": "standard",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "ok",
          "className": "button",
          "color": "primary",
          "onClick": [Function],
          "variant": "contained",
        },
        "ref": null,
        "rendered": "ok",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <TextField
            onChange={[Function]}
            placeholder=""
            required={false}
            select={false}
            variant="standard"
          />,
          <WithStyles(Button)
            className="button"
            color="primary"
            onClick={[Function]}
            variant="contained"
          >
            ok
          </WithStyles(Button)>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onChange": [Function],
            "placeholder": "",
            "required": false,
            "select": false,
            "variant": "standard",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "ok",
            "className": "button",
            "color": "primary",
            "onClick": [Function],
            "variant": "contained",
          },
          "ref": null,
          "rendered": "ok",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
